lib/include/platform/iot_platform_types_afr.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file iot_platform_types_posix.h
+ * @brief Definitions of platform layer types on POSIX systems.
+ */
+
+#ifndef _IOT_PLATFORM_TYPES_POSIX_H_
+#define _IOT_PLATFORM_TYPES_POSIX_H_
+
+#include "timers.h"
+
+
+struct iot_mutex_internal
+{
+    StaticSemaphore_t xMutex;           /**< FreeRTOS mutex. */
+    int type;                           /**< Type; used for indicating if this is reentrant or normal. */
```

Comment:
Maybe we can just use `bool recursive`, since that's what `IotMutex_Create` signature specifies.

Reply:
Agreed. I started making all 4 types posix has and then removed 2 of them as they made less sense to have.